src/SmartComponents/NoReports/index.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { createRef } from 'react';
+import { bindActionCreators } from 'redux';
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import Dropzone from 'react-dropzone';
+import {
+    Button,
+    Bullseye,
+    EmptyStateBody,
+    EmptyStateVariant,
+    EmptyState,
+    Title
+} from '@patternfly/react-core';
+// import {
+//     CubesIcon
+// } from '@patternfly/react-icons';
+import * as uploadActions from '../../actions/UploadActions';
+import { GlobalState } from '../../models/state';
+import { RouterGlobalProps } from '../../models/router';
+import ProcessImprovementSvg from '../../PresentationalComponents/Icons/process-improvement.svg';
+import './Reports.scss';
+
+interface StateToProps {
+    file: File | null
+}
+
+interface DispatchToProps {
+    selectUploadFile: (file: File) => any;
+}
+
+interface Props extends StateToProps, DispatchToProps, RouterGlobalProps {
+};
+
+interface State {
+};
+
+export class NoReports extends React.Component<Props, State> {
+
+    constructor(props: Props) {
+        super(props);
+    }
+
+    onDrop = (files: File[]) => {
+        this.props.selectUploadFile(files[0]);
+        this.props.history.push('/reports/upload');
+    };
+
+    render() {
+        const dropzoneRef: any = createRef();
+        const openFileDialog = () => {
```

Comment:
Same here, you can get rid of the Ref if you use `open` in the render function of Dropzone. Also, how come we have two places where you can upload a file? That's a little strange, right? I need to take another look at the mockups.

Reply:
You're right, I should not use ref. I've corrected this problem in my last commit.